pallets/xcmp-handler/src/tests.rs

```diff
@@ -128,3 +128,64 @@ fn errors_if_not_found() {
 		);
 	});
 }
+
+//*****************
+//Helper  functions
+//*****************
+
+// calculate_xcm_fee
+#[test]
+fn returns_fee() {
+	new_test_ext().execute_with(|| {
+		let currency_id = CurrencyId::Native;
+		let para_id: u32 = 1000;
+		let xcm_data = XcmCurrencyData {
+			native: false,
+			fee_per_second: 50_000_000_000,
+			instruction_weight: 1_000,
+		};
+		XcmChainCurrencyData::<Test>::insert(para_id, currency_id, xcm_data.clone());
+		let transact_encoded_call_weight: u64 = 100_000_000;
+
+		let expected_fee = xcm_data.fee_per_second * (transact_encoded_call_weight as u128) /
+			(WEIGHT_PER_SECOND as u128);
```

Comment:
It'd be nice if this were just the static value for simplicity, but I guess `WEIGHT_PERSECOND` could potentially change and is non-trivial to stub.